### ISSUE TITLE
Some buttons on the header are tagged with the `tab` role without an associated element with the `tabpanel` role

### ISSUE THREAD

**Issue by luixxiul** [vector-im/element-web#25016]
### Steps to reproduce

- Create E2E test to select buttons inside `HeaderButtons` on the room header such as the thread button and the room info button with `findByRole("button", {name: "" })`.

#### What did you expect?

The buttons should be able to be found.

#### What happened instead?

They were not found  as they are tagged as [`tab`](https://github.com/matrix-org/matrix-react-sdk/blob/24eed963d3018e3dc6a23f578d9e11e4048872e7/src/components/views/right_panel/HeaderButton.tsx#L58).

Since the content rendered by clicking those buttons is not a tabbed content tagged with `role="tabpanel"`, it should be fine to tag them as `role="button"`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#description

I assume the `role="tab"` rather should be applied to the left column buttons on the setting dialogs, for example.

### Will you send logs?

No

**Comment by andybalaam** [comment]
Are you talking about this button?

![image](https://user-images.githubusercontent.com/76812/229449799-11092fad-bed4-422c-a25c-4430d93cbe4e.png)

Why is it a problem? Isn't there another way to find this button?

**Comment by luixxiul** [comment]
I forgot that [`findbutton()`](https://github.com/matrix-org/matrix-react-sdk/blob/4ed6e39067955f2430f8506086127cb9486f1f9a/cypress/support/find.ts#L53-L55) was a function based on Cypress Testing Library, which can be used like this: [`cy.get(".mx_EventTile_last").realHover().findButton("Edit").click();`](https://github.com/matrix-org/matrix-react-sdk/pull/10480/files#diff-61341016794840db0e9ca539fbd11562023a1da08052fdc8f8781972df746071R49)

Currently it is not possible to find the button with `cy.get(".mx_RoomHeader").findButton("Threads").click();` as the button's role is specified as `role="tab"`, while `findButton()` seeks an element with the role "button".

**Comment by andybalaam** [comment]
I'm not convinced that we should change the button's role just because it makes the test look nicer. There might be lots of reasons why it is like this, or unintended consequences of changing it.

How about adding a new function `findTabButton` that finds things with `role="tab"`?

**Comment by luixxiul** [comment]
I see, I am going to add `findTab` function to `find.ts`, but I am not quite sure whether `role="tab"` on those buttons would be still effective.

According to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#description,

> The ARIA `tab` role indicates an interactive element ... displays its associated `tabpanel`.

> An element with the `tab` role controls the visibility of an associated element with the `tabpanel` role.

but I could not find an element tagged with `tabpanel` role, other than on emojipicker's [`Category.tsx`](https://github.com/matrix-org/matrix-react-sdk/blob/5cd459ff2d648f143d0a9c24f96f89aca40616c8/src/components/views/emojipicker/Category.tsx#L99) and [`ThirdPartyMemberInfo.tsx`](https://github.com/matrix-org/matrix-react-sdk/blob/5cd459ff2d648f143d0a9c24f96f89aca40616c8/src/components/views/rooms/ThirdPartyMemberInfo.tsx#L150). 

[Here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#example) is an example which combines the role `tab` with `tablist` and elements with `tabpanel` to create an interactive group of tabbed content:

![1](https://user-images.githubusercontent.com/3362943/229465975-bcc52ea8-a022-47dd-b0eb-a81872039f13.png)

I suspect `BaseCard` has been regarded as `tabpanel`, though it has not been coded so.

**Comment by luixxiul** [comment]
I'm going to change the title to reflect the actual issue (which is not really about a test; sorry for confusion).

**Comment by andybalaam** [comment]
Thank you, I think this is definitely worth thinking about, and in the meantime if you add `findTab` (though I think I prefer `findTabButton` ;-) then it won't block your tests.

**Comment by luixxiul** [comment]
I understood, thanks!
